File: api/app/Recipe.php
<?php

namespace App;

abstract class Recipe {
    abstract public function title(): string;
    abstract public function time(): string;
    abstract public function servings(): string;
    abstract public function ingredients(): string;
    abstract public function instructions(): string;
    abstract public function image(): string;
    abstract public function url(): string;

    public function __get(string $name)
    {
        return $this->{$name}();
    }
}

File: internal/data/recipe.go
package data

import "time"

type Recipe struct {
	ID           int64     `json:"id"`
	Title        string    `json:"title"`
	Time         int32     `json:"time"`
	Servings     int32     `json:"servings"`
	Ingredients  []string  `json:"ingredients"`
	Instructions []string  `json:"instructions"`
	Image        string    `json:"image"`
	Url          string    `json:"url"`
	CreatedAt    time.Time `json:"-"`
	UpdatedAt    time.Time `json:"-"`
}
